src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -29,41 +44,54 @@ public static void QuoteTableAndColumns(Configuration configuration)
 			}
 		}
 
-		private static ISet<string> GetReservedWords(IDictionary<string, string> cfgProperties)
+		private static ISet<string> GetReservedWords(Configuration configuration, Dialect.Dialect dialect)
 		{
-			var dialect = Dialect.Dialect.GetDialect(cfgProperties);
-			var connectionHelper = new ManagedProviderConnectionHelper(cfgProperties);
-			return GetReservedWords(dialect, connectionHelper);
+			IConnectionHelper connectionHelper = new ManagedProviderConnectionHelper(configuration.GetDerivedProperties());
+			connectionHelper.Prepare();
+			try
+			{
+				return GetReservedWords(dialect, connectionHelper);
+			}
+			finally
+			{
+				connectionHelper.Release();
+			}
 		}
 
-		private static ISet<string> GetReservedWords(Dialect.Dialect dialect, IConnectionHelper connectionHelper)
+		private static ISet<string> GetReservedWords(IConnectionProvider connectionProvider, Dialect.Dialect dialect)
 		{
-			ISet<string> reservedDb = new HashSet<string>();
+			IConnectionHelper connectionHelper = new SuppliedConnectionProviderConnectionHelper(connectionProvider);
 			connectionHelper.Prepare();
 			try
 			{
-				var metaData = dialect.GetDataBaseSchema(connectionHelper.Connection);
-				foreach (var rw in metaData.GetReservedWords())
-				{
-					reservedDb.Add(rw.ToLowerInvariant());
-				}
+				return GetReservedWords(dialect, connectionHelper);
 			}
 			finally
 			{
 				connectionHelper.Release();
 			}
-			return reservedDb;
+		}
+
+		private static ISet<string> GetReservedWords(Dialect.Dialect dialect, IConnectionHelper connectionHelper)
+		{
+			ISet<string> reservedWords = new HashSet<string>(dialect.Keywords, StringComparer.OrdinalIgnoreCase);
```

Comment:
This HashSet is completely useless:

1. We create this hash set from `dialect.Keywords`.
2. We add another hash set from `metaData.GetReservedWords()`.
3. We merge this hash set into `dialect.Keywords`;